server/src/services/plugin-execution.service.ts

```diff
@@ -0,0 +1,214 @@
+import { Plugin as ExtismPlugin, newPlugin } from '@extism/extism';
+import { Injectable } from '@nestjs/common';
+import { resolve } from 'node:path';
+import { PLUGIN_JWT_SECRET } from 'src/constants';
+import { Asset, WorkflowAction, WorkflowFilter } from 'src/database';
+import { OnEvent, OnJob } from 'src/decorators';
+import { JobName, JobStatus, QueueName } from 'src/enum';
+import { ArgOf } from 'src/repositories/event.repository';
+import { PluginContext, pluginTriggers, PluginTriggerType } from 'src/schema/tables/plugin.table';
+import { BaseService } from 'src/services/base.service';
+import { PluginHostFunctions } from 'src/services/plugin-host.functions';
+import { IWorkflowJob, JobItem, JobOf, WorkflowData } from 'src/types';
+import { TriggerConfig } from 'src/types/plugin-schema.types';
+
+interface WorkflowContext {
+  jwtToken: string;
+  asset: Asset;
+  triggerConfig: TriggerConfig | null;
+}
+
+@Injectable()
+export class PluginExecutionService extends BaseService {
+  private loadedPlugins: Map<string, ExtismPlugin> = new Map();
+  private hostFunctions!: PluginHostFunctions;
+
+  @OnEvent({ name: 'AppBootstrap' })
+  async onBootstrap() {
+    this.hostFunctions = new PluginHostFunctions(
+      this.assetRepository,
+      this.albumRepository,
+      this.accessRepository,
+      this.cryptoRepository,
+      this.logger,
+    );
+    await this.loadCorePlugins();
+  }
+
+  private async loadCorePlugins() {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    for (const plugin of plugins) {
+      try {
+        const pluginPath = resolve(process.cwd(), '..', plugin.wasmPath);
```

Comment:
you can't have file system access like this in a service. even in a repository it cannot be using `process.cwd()` to resolve anything.